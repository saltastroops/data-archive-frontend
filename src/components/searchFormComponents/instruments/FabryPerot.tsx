import * as React from "react";
import MultiSelectField, {
  AllOption
} from "../../basicComponents/MultiSelectField";

interface IFabryPerotProps {
  fabryPerot?: string[];
  onSelect: (value: any) => void;
}

export const FABRY_PEROT = [
  "Low resolution",
  "Medium resolution",
  "High resolution",
  "Tunable filter"
];
/**
 * A form for selecting FabryPerot-related search parameters.
 */
const FabryPerot = (props: IFabryPerotProps) => {
  const { onSelect, fabryPerot } = props;

  // Function for handling change events
  const select = (e: React.SyntheticEvent<HTMLSelectElement>) => {
    const values = e.currentTarget.selectedOptions;
    onSelect({
      fabryPerot: Array.from(values).map((t: any) => {
        return t.text;
      })
    });
  };

  return (
    <div>
      <p>Fabry perot</p>
      <MultiSelectField
        name={"fabryPerot"}
        onChange={select}
        value={fabryPerot || ["All"]}
      >
        <AllOption />
        {FABRY_PEROT.sort().map(i => (
          <option key={i} value={i}>
            {i}
          </option>
        ))}
      </MultiSelectField>
    </div>
  );
};

export default FabryPerot;
