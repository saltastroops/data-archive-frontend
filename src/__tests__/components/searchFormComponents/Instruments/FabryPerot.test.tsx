import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import FabryPerot from "../../../../components/searchFormComponents/instruments/FabryPerot";

const onSelect = jest.fn() as any;
describe("FabryPerot", () => {
  it("should be defined", () => {
    expect(
      mount(<FabryPerot fabryPerot={[]} onSelect={onSelect} />)
    ).toBeDefined();
  });

  it("should render correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(mount(<FabryPerot fabryPerot={[]} onSelect={onSelect} />))
    ).toMatchSnapshot();
  });
});
