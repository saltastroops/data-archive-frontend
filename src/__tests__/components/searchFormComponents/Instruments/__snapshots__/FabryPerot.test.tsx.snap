// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FabryPerot should render correctly 1`] = `
.c0 {
  width: 95%;
}

<FabryPerot
  fabryPerot={Array []}
  onSelect={[MockFunction]}
>
  <div>
    <p>
      Fabry perot
    </p>
    <MultiSelectField
      name="fabryPerot"
      onChange={[Function]}
      value={Array []}
    >
      <styled.div
        className="control select is-multiple "
      >
        <StyledComponent
          className="control select is-multiple "
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  width: 95%;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="control select is-multiple  c0"
          >
            <select
              multiple={true}
              name="fabryPerot"
              onChange={[Function]}
              size={6}
              style={
                Object {
                  "width": "216px",
                }
              }
              value={Array []}
            >
              <AllOption>
                <option
                  value="All"
                >
                  All
                </option>
              </AllOption>
              <option
                key="High resolution"
                value="High resolution"
              >
                High resolution
              </option>
              <option
                key="Low resolution"
                value="Low resolution"
              >
                Low resolution
              </option>
              <option
                key="Medium resolution"
                value="Medium resolution"
              >
                Medium resolution
              </option>
              <option
                key="Tunable filter"
                value="Tunable filter"
              >
                Tunable filter
              </option>
            </select>
          </div>
        </StyledComponent>
      </styled.div>
    </MultiSelectField>
  </div>
</FabryPerot>
`;
